src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -0,0 +1,216 @@
+       Designer Assembly, for backward compatability.
+   -->
+   <GenerateResourceDesignerIntermediateClass
+      Namespace="$(AndroidResgenNamespace)"
+      IsApplication="$(AndroidApplication)"
+      OutputFile="$(_GenerateResourceDesignerClassFile)"
+   >
+   </GenerateResourceDesignerIntermediateClass>
+   <Touch Files="$(_GenerateResourceDesignerClassFile)" />
+   <ItemGroup>
+     <FileWrites Include="$(_GenerateResourceDesignerClassFile)" />
+   </ItemGroup>
+</Target>
+
+<Target Name="_GenerateResourceDesignerAssembly"
+    DependsOnTargets="_GenerateResourceCaseMap"
+    Inputs="$(_DesignerIntermediateOutputPath)R.txt"
+    Outputs="$(_GenerateResourceDesignerAssemblyOutput)"
+    Condition=" '$(AndroidUseDesignerAssembly)' == 'True' And '$(DesignTimeBuild)' == 'True' ">
+  <GenerateResourceDesignerAssembly
+      ContinueOnError="$(DesignTimeBuild)"
+      RTxtFile="$(_DesignerIntermediateOutputPath)R.txt"
+      IsApplication="$(AndroidApplication)"
+      DesignTimeBuild="$(DesignTimeBuild)"
+      Deterministic="$(Deterministic)"
+      OutputFile="$(_GenerateResourceDesignerAssemblyOutput)"
+      TargetFrameworkIdentifier="$(TargetFrameworkIdentifier)"
+      TargetFrameworkVersion="$(TargetFrameworkVersion)"
+      ProjectDir="$(ProjectDir)"
+      Resources="@(AndroidResource);@(AndroidBoundLayout)"
+      ResourceDirectory="$(MonoAndroidResourcePrefix)"
+      CaseMapFile="$(_GenerateResourceCaseMapFile)"
+      FrameworkDirectories="$(_XATargetFrameworkDirectories);$(_XATargetFrameworkDirectories)Facades"
+      AdditionalResourceDirectories="@(LibraryResourceDirectories)"
+      AssemblyName="$(AssemblyName)"
+  >
+  </GenerateResourceDesignerAssembly>
+  <ItemGroup>
+    <FileWrites Include="$(_GenerateResourceDesignerAssemblyOutput)" />
+  </ItemGroup>
+</Target>
+
+<Target Name="_CheckForInvalidDesignerConfig"
+    DependsOnTargets="_AddAndroidCustomMetaData"
+    Condition=" '$(AndroidUseDesignerAssembly)' != 'True' ">
+  <ItemGroup>
+    <_ErrorItems Include="@(_MonoAndroidReferencePath)" Condition=" '%(_MonoAndroidReferencePath.HasResourceDesignerAssemblyReference)' == 'True' "/>
+    <_ErrorItems Include="@(_MonoAndroidReferenceDependencyPaths)"  Condition=" '%(_MonoAndroidReferenceDependencyPaths.HasResourceDesignerAssemblyReference)' == 'True' "/>
+  </ItemGroup>
+  <AndroidError Code="XA1031"
+      ResourceName="XA1031"
+      FormatArguments="%(_ErrorItems.FullPath)"
+      Condition=" '@(_ErrorItems->Count())' != '0' "
+  />
+</Target>
+
+<PropertyGroup>
+  <_BuildResourceDesignerDependsOn>
+    _SetupDesignerProperties;
+    _GenerateResourceCaseMap;
+    _GenerateRtxt;
+    _GenerateResourceDesignerIntermediateClass;
+    _GenerateResourceDesignerAssembly;
+    _AddResourceDesignerFiles;
+  </_BuildResourceDesignerDependsOn>
+</PropertyGroup>
+
+<Target Name="_AddResourceDesignerFiles"
+    DependsOnTargets="_SetupDesignerProperties"
```

Comment:
…on the "sort properties alphabetically vs. have an ordering for clarity" front, *I* prefer having `Condition` as the first attribute most (all?) of the time, because when debugging later that's typically the most important thing I care about, a'la:

```xml
<Target Name="_AddResourceDesignerFiles"
    Condition=" '$(AndroidUseDesignerAssembly)' == 'True' "
    DependsOnTargets="_SetupDesignerProperties"
    AfterTargets="GenerateGlobalUsings">
  …
```